test/k8sT/Updates.go

```diff
@@ -280,3 +304,53 @@ func InstallAndValidateCiliumUpgrades(kubectl *helpers.Kubectl, oldVersion, newV
 	}
 	return testfunc, cleanupCallback
 }
+
+// patchKubeDNSWithFixedIdentity patches the kube-dns with fixed identity
+// labels and returns the labels stored in the deployment template before
+// applying the patch.
+func patchKubeDNSWithFixedIdentity(kubectl *helpers.Kubectl) (map[string]string, error) {
+	dnsDeployment := "deployment/kube-dns"
+	switch helpers.GetCurrentK8SEnv() {
+	case "1.11", "1.12":
+		dnsDeployment = "deployment/coredns"
+	}
+	fb, err := kubectl.Get(helpers.KubeSystemNamespace, dnsDeployment).Filter("{.spec.template.metadata.labels}")
+	if err != nil {
+		return nil, err
+	}
+
+	oldLabelsMap := fb.KVOutput()
+
+	// Apply fixed identity patch
+	cmdRes := kubectl.Exec(fmt.Sprintf(`kubectl patch -n kube-system %s --type merge -p '{"spec":{"template":{"metadata":{"labels":{"io.cilium.fixed-identity":"kube-dns"}}}}}'`,
+		dnsDeployment))
+	if !cmdRes.WasSuccessful() {
+		return nil, fmt.Errorf("%s", cmdRes.OutputPrettyPrint())
+	}
+	return oldLabelsMap, nil
+}
+
+// revertPatchKubeDNSWithFixedIdentity reverts the changes made by
+// PatchKubeDNSWithFixedIdentity
+func revertPatchKubeDNSWithFixedIdentity(kubectl *helpers.Kubectl, oldLabels map[string]string) error {
+	dnsDeployment := "deployment/kube-dns"
```

Comment:
I would move this, to something helpers.GetDNSDeploymentName() similar as `DNSDeployment()`
```
	dnsDeployment := "deployment/kube-dns"
	switch helpers.GetCurrentK8SEnv() {
	case "1.11", "1.12":
		dnsDeployment = "deployment/coredns"
	}
```